pkg/machine/wsl/machine.go

```diff
@@ -0,0 +1,1055 @@
+// +build amd64 arm64
+
+package wsl
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/containers/podman/v3/pkg/machine"
+	"github.com/containers/podman/v3/utils"
+	"github.com/containers/storage/pkg/homedir"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/text/encoding/unicode"
+	"golang.org/x/text/transform"
+)
+
+var (
+	// vmtype refers to qemu (vs libvirt, krun, etc)
+	vmtype = "wsl"
+)
+
+const (
+	ErrorSuccessRebootInitiated = 1641
+	ErrorSuccessRebootRequired  = 3010
+)
+
+const containersConf = `[containers]
+netns="slirp4netns"
```

Comment:
Why? This should not be set so it can use the default if possible.

Reply:
I should totally have commented my reasoning here, sorry for it being buried in the PR. The reasons for this are

1. I ran into nftables compatibility issues with some combinations of the WSL Linux Kernel maintained by Microsoft and some versions of Linux distributions. This is not the case atm with latest Fedora and WSL, but since they aren't synced together anything deeply connected to kernel facilities can have problems. 
2. All WSL instances, including multiple podman machine instances and any additional linux distribution instances users have installed all run together under the same kernel, with the same network interfaces, etc. So there is a strong potential for conflict, one linux distro might overwrite or augment netfilter policy. 

So in short, usermode networking is more robust and felt safer as a default. On the other hand compatibility is impacted with container to container communication. So we could decide to attempt bridge and if it starts to lead to problems go back to a slirp config. 

Let me know what you guys think.

Reply:
@rhatdan any opinion on ^

Reply:
@baude @vrothberg @giuseppe @mheon WDYT?
I can understand both sides.  But I rely on @Luap99 and others for network setups.

Reply:
@Luap99 based on the above wdyt we should do. I'm happy to try bridge/defaults for now

Reply:
If the default works I would like to use it. Many users expect the inter container communication.

Reply:
@Luap99 sounds good I will back that part of the change out shortly